packages/connect/src/api/ethereum/api/ethereumSignTransaction.ts

```diff
@@ -118,6 +120,71 @@ export default class EthereumSignTransaction extends AbstractMethod<
         return getNetworkLabel('Sign #NETWORK transaction', this.params.network);
     }
 
+    async payloadToPrecomposed() {
+        const feePerByte = new BigNumber(
+            this.payload.transaction.gasPrice || this.payload.transaction.maxFeePerGas!,
+        );
+        const fee = feePerByte.multipliedBy(this.payload.transaction.gasLimit);
+        const data = this.payload.transaction.data?.replace(/^0x/, '');
+        let recipient = this.payload.transaction.to!;
+        let amount = new BigNumber(this.payload.transaction.value);
+        let totalSpent = amount.plus(fee);
+        let token: TokenInfo | undefined;
+
+        // ERC-20 transfer
+        if (this.payload.transaction.to && data?.startsWith('a9059cbb') && amount.eq(0)) {
```

Comment:
just a thought that crossed my mind - maybe there are some eth utils already in one place such as this one https://github.com/trezor/trezor-suite/blob/feat/suite-connect-sign-modals/suite-common/wallet-utils/src/sendFormUtils.ts#L114 (which obviously can't be imported here) and maybe we want to have various eth utils in one place and reuse them. 

totally not a request for anything

Reply:
Yes, I was looking at this code, but we would have to move it to a different package. And this particular function basically does the opposite